src/util/proto.ts

```diff
@@ -1,32 +1,39 @@
 import * as Pb from '../_generated/proto/satellite'
 import * as _m0 from 'protobufjs/minimal';
 
-let msgtypemapping: { [k: string]: [number, any] } = {
-    "Electric.Satellite.SatErrorResp": [0, Pb.SatErrorResp],
-    "Electric.Satellite.SatAuthReq": [1, Pb.SatAuthReq],
-    "Electric.Satellite.SatAuthResp": [2, Pb.SatAuthResp],
-    "Electric.Satellite.SatGetServerInfoReq": [3, Pb.SatGetServerInfoReq],
-    "Electric.Satellite.SatGetServerInfoResp": [4, Pb.SatGetServerInfoResp],
-    "Electric.Satellite.SatPingReq": [5, Pb.SatPingReq],
-    "Electric.Satellite.SatPingResp": [6, Pb.SatPingResp],
-    "Electric.Satellite.SatInStartReplicationReq": [7, Pb.SatInStartReplicationReq],
-    "Electric.Satellite.SatInStartReplicationResp": [8, Pb.SatInStartReplicationResp],
-    "Electric.Satellite.SatInStopReplicationReq": [9, Pb.SatInStopReplicationReq],
-    "Electric.Satellite.SatInStopReplicationResp": [10, Pb.SatInStopReplicationResp],
-    "Electric.Satellite.SatOpLog": [11, Pb.SatOpLog],
-    "Electric.Satellite.SatRelation": [12, Pb.SatRelation],
-    "Electric.Satellite.SatMigrationNotification": [13, Pb.SatMigrationNotification]
-};
+// NOTE: This mapping should be kept in sync with Electric message mapping.
+// Take into account that this mapping is dependent on the protobuf
+// protocol version.
+let msgtypetuples: { [k: string]: [number, any] } =
```

Comment:
Suggestion: it's easy to forget to change/add/make typo in the mapping here, however you're unlikely to forget adding it to SatPbMsg since you want to use it. Let's make it safer for us to change in one of two ways (both imply moving `export type SatPbMsg = ...` before this line :

1. We can just ensure that we've specified a mapping for every one of the names:
    ```typescript
    type Names = SatPbMsg["$type"] extends `Electric.Satellite.v0_1.${infer K}` ? K : never
    
    let msgtypetuples: { [k in Names]: [number, any] } = { ... }
    ```
    This way we get errors on typos ("unknown property specified") and on missing properties
2. We can go a step further and ensure that we're mapping correct values:
    ```typescript
    export type SatPbMsgObj<Type extends string = string> = {
        $type: Type;
        encode(message: SatPbMsg, writer: _m0.Writer): _m0.Writer;
        decode(input: _m0.Reader | Uint8Array, length?: number): SatPbMsg;
        fromPartial<I extends Pb.Exact<Pb.DeepPartial<SatPbMsg>, I>>(object: I): SatPbMsg;
    };
    
    type GetName<T extends {"$type": string}> = T["$type"] extends `Electric.Satellite.v0_1.${infer K}` ? K : never
    type MappingTuples = {[k in SatPbMsg as GetName<k>]: [number, SatPbMsgObj<k['$type']>]}
    
    let msgtypetuples: MappingTuples = { ... }
    ```
    
    This catches errors like `"SatAuthResp": [2, Pb.SatPingReq],` but the code is more complex and a little less flexible

Reply:
Great suggestion, I will personally vote for the second option.

Reply:
Yeah, me too, just wasn't sure if you'd like the type complexity